src/darwin/Framework/CHIP/MTRDeviceController.mm

```diff
@@ -449,6 +453,53 @@ - (BOOL)setupCommissioningSessionWithPayload:(MTRSetupPayload *)payload
     return [self syncRunOnWorkQueueWithBoolReturnValue:block error:error];
 }
 
+- (BOOL)setupCommissioningSessionWithDiscoveredDevice:(MTRCommissionableBrowserResult *)discoveredDevice
+                                              payload:(MTRSetupPayload *)payload
+                                            newNodeID:(NSNumber *)newNodeID
+                                                error:(NSError * __autoreleasing *)error
+{
+    auto block = ^BOOL {
+        chip::NodeId nodeId = [newNodeID unsignedLongLongValue];
+        self->_operationalCredentialsDelegate->SetDeviceID(nodeId);
+
+        auto errorCode = CHIP_ERROR_INVALID_ARGUMENT;
+        if (discoveredDevice.params.HasValue()) {
+            auto params = discoveredDevice.params.Value();
+            auto pinCode = static_cast<uint32_t>([[payload setupPasscode] unsignedLongValue]);
+            params.SetSetupPINCode(pinCode);
+
+            errorCode = self.cppCommissioner->EstablishPASEConnection(nodeId, params);
+        } else {
+            // Try to get a QR code if possible (because it has a better
+            // discriminator, etc), then fall back to manual code if that fails.
+            NSString * pairingCode = [payload qrCodeString:nil];
+            if (pairingCode == nil) {
+                pairingCode = [payload manualEntryCode];
+            }
+            if (pairingCode == nil) {
+                return ![MTRDeviceController checkForError:CHIP_ERROR_INVALID_ARGUMENT logMsg:kErrorSetupCodeGen error:error];
+            }
+
+            for (id key in discoveredDevice.interfaces) {
+                auto resolutionData = discoveredDevice.interfaces[key].resolutionData;
+                if (!resolutionData.HasValue()) {
+                    continue;
+                }
+
+                errorCode = self.cppCommissioner->EstablishPASEConnection(
```

Comment:
Hmm.  So we had a discovered instance name, but not an actual IP/port for it, and we want to go ahead and use that instance name instead of redoing the discovery?  Why?  At the very least this needs documentation.

Reply:
We are actually passing a `CommonResolutionData` struct that contains both the port and a variable number of IPs. This is the reason why we are not doing discovery again.

Reply:
So we are guaranteed that if we got here then the actual AAAA lookups and whatnot succeeded?  OK.